packages/core/src/codewhispererChat/tools/fsWrite.ts

```diff
@@ -108,6 +102,20 @@ export class FsWrite {
         return diffLines(oldContent, newContent)
     }
 
+    public async getBackup(): Promise<FsWriteBackup> {
+        const sanitizedPath = sanitizePath(this.params.path)
+        let oldContent
+        let isNew
+        try {
+            oldContent = await fs.readFileText(sanitizedPath)
+            isNew = false
+        } catch (err) {
+            oldContent = ''
+            isNew = true
+        }
+        return { filePath: sanitizedPath, content: oldContent, isNew }
```

Comment:
Is the logic here that for backups that don't exist (i.e. writing to a new file) we return empty content and a flag to say it doesn't exist? Is this the easiest way to handle this case?

Reply:
Yeah, we need some way to differentiate if the backup was a new file creation or the file existed but was empty. I think a flag works best here